api/backup.go

```diff
@@ -1,24 +1,130 @@
 package api
 
 import (
+	"fmt"
 	"net/http"
+	"strconv"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/gorilla/mux"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/util"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 
 	"github.com/rancher/go-rancher/api"
 	"github.com/rancher/go-rancher/client"
 )
 
+const (
+	BackupTargetDefaultPollInterval = 300
+)
+
 func (s *Server) BackupTargetList(w http.ResponseWriter, req *http.Request) error {
 	apiContext := api.GetApiContext(req)
 
-	backupTargets, err := s.m.ListBackupTargetsSorted()
+	bts, err := s.backupTargetList()
+	if err != nil {
+		return err
+	}
+	apiContext.Write(bts)
+	return nil
+}
+
+func (s *Server) backupTargetList() (*client.GenericCollection, error) {
+	bts, err := s.m.ListBackupTargetsSorted()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list backup targets")
+	}
+	return toBackupTargetCollection(bts), nil
+}
```

Comment:
Why separate this function  while Get, Create, Update don't?

Reply:
Just follow the other apis such as the recurring job.
But it's okay to be original, and modified. Thanks.